Make clean is not working with generated files

**What seems to be the problem?**
I have a simple premake5 file that creates makefiles for a small project of mine. When I run premake5 gmake2 all makefiles are created successfully and I can call make to compile to an executable and run it.
The problem is when I try to call make clean, I get an error saying: `The directory name is invalid. and make: *** [clean] Error 267`

**What did you expect to happen?**
Build directory should have been cleaned

**What have you tried so far?**
I've narrowed it down to one line in the makefile which should remove all the object files, but causes an error and stops any more files from being removed.

`$(SILENT) if exist $(subst /,\\,$(GENERATED)) rmdir /s /q $(subst /,\\,$(GENERATED))`

This is the line that is causing the error. Removing it fixes the problem but causes my bin/obj folder to remain.

**How can we reproduce this?**
Calling make clean gives this error.

**What version of Premake are you using?**
premake5 (Premake Build Script Generator) 5.0.0-alpha16.

**Anything else we should know?**
This issue is based on https://stackoverflow.com/questions/64414270/make-clean-is-not-working-when-using-premake5
though this problem occurs with me too.

I'm sorry for the poor explanation. I'm super new to this.
I'm using MinGW-w64  for windows downloaded from https://winlibs.com
Just by running the command  after building the project
`mingw32-make clean`

This error occures

```
The directory name is invalid.
mingw32-make[1]: *** [Makefile:101: clean] Error 267
mingw32-make: *** [makefile:42: clean] Error 2
```
and this error and this seems to only happenes when I have only one cpp file in my project.
This was not the problem in `premake5 (Premake Build Script Generator) 5.0.0-alpha14`

Ok, here is a simple project that reproduces the error for me

```
|-- Build/
|   |-- premake5.exe
|
|-- Source/
|   |-- Main.cpp
|-- premake5.lua
```

Main.cpp
```
#include <iostream>

int main()
{
	std::cout << "Hello World!\n";
}
```
premake5.lua

```
workspace "Project-Workspace"
	configurations
	{
		"Debug",
		"Release"
	}

project "Project"
	kind "ConsoleApp"
	language "C++"
	targetdir "Binaries/%{cfg.buildcfg}/%{prj.name}"
	objdir "Binaries/%{cfg.buildcfg}/%{prj.name}/Objects"

	files
	{
		"Source/Main.cpp"
	}

	filter "configurations:Debug"	
		symbols "on"
		runtime "Debug"

	filter "configurations:Release"
		runtime "Release"
		optimize "on"
```
then I run

`Build\premake5.exe gmake2`
`mingw32-make`

The project files are generated and builds just fine, but after that if  I run

`mingw32-make clean`

it results in an error saying 
```
Cleaning Project
The directory name is invalid.
mingw32-make[1]: *** [Project.make:99: clean] Error 267
mingw32-make: *** [makefile:34: clean] Error 2
```

--- a/modules/gmake2/gmake2_cpp.lua
+++ b/modules/gmake2/gmake2_cpp.lua
@@ -690,7 +690,7 @@
 		_p('\t$(SILENT) rm -rf $(OBJDIR)')
 		_p('else')
 		_p('\t$(SILENT) if exist $(subst /,\\\\,$(TARGET)) del $(subst /,\\\\,$(TARGET))')
-		_p('\t$(SILENT) if exist $(subst /,\\\\,$(GENERATED)) rmdir /s /q $(subst /,\\\\,$(GENERATED))')
+		_p('\t$(SILENT) if exist $(subst /,\\\\,$(GENERATED)) del /s /q $(subst /,\\\\,$(GENERATED))')
 		_p('\t$(SILENT) if exist $(subst /,\\\\,$(OBJDIR)) rmdir /s /q $(subst /,\\\\,$(OBJDIR))')
 		_p('endif')
 		_p('')

